docker.makefile

```diff
@@ -0,0 +1,57 @@
+#Dockerfile vars
```

Comment:
@mlxd  agreed with docker-compose, docker-compose is more robust, But I believe Makefile implementation should also be there in case someone wants to do things  using Kubernetes or  just doing build-testing for the local ENV only.

Reply:
Thanks @arshpreetsingh While I agree that the makefile implementation is definitely useful, the issue is more for cross-platform use as they can only be used on Linux/Mac machines --- I would also expect deploying to a Kubernetes cluster to be more in-line with the docker-dompose syntax, given that both are configured in YAML, though I think Kubernetes support can come a little after Docker support in a future PR :slightly_smiling_face:  

If we intend to add support for Windows, we should ensure the users have a way to access this functionality on a Windows machine, or really any machine on which Docker supports.

Also, I think it would be good even with the makefile/docker-compose.yml, to list the commands required to access the PennyLane image as a runnable environment using docker: something like `docker run -it pennylane:latest /bin/bash` for those not familiar with running docker containers.

This gives us a few potentially different options to explore:
- Keep the makefile, and add a comment to the README to use the makefile with WSL under Windows if Docker support is required.
- Keep the makefile and add an equivalent Powershell file for building the images on Windows, with similar functionality as the makefile.
- Replace the makefile with a cross-platform docker-compose.yml file (also allows easy building, running and mounting of the cloned PL repository into the image).
- Explicitly list the build & run commands on the README for each given image the user wishes to use.

For now, I think the quickest change may be to add the comments to the README for using WSL under Windows.

Reply:
@mlxd  So should we talk about docker-compose implementation  in this PR or in other one?